### ISSUE TITLE
[Bug] Spells Seem to be Distributed Based on Their Vanilla Chests

### ISSUE THREAD

**Issue by jet082** [abyssonym/sai2_rando#4]
I believe that the spells may be tied to their chests somehow. I was watching someone stream this and they went from not having the light spell to having it shortly after checking the vanilla chest.

See the clip below, where you can tell what spells he has on the menu.

https://www.youtube.com/watch?v=-U7eU_bC9e4

**Comment by Jigglysaint** [comment]
I can confirm that you do indeed get the Light Spell when you open the light spell chest with anything else inside, but the same doesn't worth for all the spell chests.  I tried with Moon Spell and nothing.

**Comment by jet082** [comment]
I believe I might know a bit more about this now. The "light spell" chest is, I believe, an "event chest" and not a standard chest. It will be already opened if you have the light spell going into the room in the vanilla game (which is different from the non-event chests, which have their own open/closed flags in RAM). Bafflingly, the aqua and star spells seem to be standard chests.

However, it is currently listed in the tables/chest_pointers.txt file instead of the table/event_chest_pointers.txt file.

Additionally, its listing in the tables/chest_pointers.txt is "thunderspell," despite its listing in tables/requirements.txt being "lightspell."

Maybe fixing these lines would be enough to fix this bug?

**Comment by abyssonym** [comment]
The name there doesn't actually impact anything, that's just for my own reference.

**Comment by abyssonym** [comment]
There's a couple of conditionals that check for the light spell at `37ab0` and `37af6`. These are probably for entering Fuwa Fuwa island. Other than that there's no events in the rom that match the pattern of the other event chests, according to my tools. And there's a non-event chest that matches the pattern at `1911d`, so I'm not sure what's going on here.

**Comment by Jigglysaint** [comment]
Okay I found the issue.  As it turns out, the 7A 70 04 and the 7B 70 04 are doing the exact same thing.  When I changed 7A 70 04 to 7A 71 04, it gave me the star spell instead.

Okay so here is how it works.  I changed 7B 70 04 to 7B 69 04, which is the elven flute.  Then I changed 7A 70 04 to 7A 4E 04, which means that if I obtained the Silver Sword, the chest will be empty.  This should work for every chest that is not a boss drop, but it does mean that the game also keeps track of the contents of chests using separate flags.  This explains why you can't seem to move bottles or money into "event chests".
In short:  "Event chests" are using their own inventory bit as a chest gatekeeper, equipment chests and upgrades use a different list of bits as chest gatekeepers.  Boss chests of course are still very different.

**Comment by abyssonym** [comment]
Ah, of course. I did notice before that ordinary chests indicated which memory flag to use after the `7a` byte. However, I assumed that all of these chests used non-item addresses for that purpose. That seems to be true in the majority of cases, but this exception bit me in the ass. The solution is to change the `7a` and `7b` values to be the same.

**Comment by abyssonym** [comment]
(although by using that solution we can't put generic items in that chest, so if possible it might be better to find an unused address and use that)

**Comment by Jigglysaint** [comment]
Each generic item should already have an address, so all you would need to do is shuffle that.  The ID for the Silver Sword chest is 90 04.  91 04 is used for the first life bottle.  I also don't know if setting flags that were never part of the ram will be saved to the ram.  Otherwise you could use those instead.

**Comment by abyssonym** [comment]
Each generic chest already has an address, except for the light spell chest, which uses the "light spell obtained" address instead. That's why I'd have to find a new one.

**Comment by abyssonym** [comment]
Well, I can probably free up some addresses from generic chests that now contain unique items. But currently I don't do that.

**Comment by jet082** [comment]
So how possible is this to fix, do you think?

**Comment by abyssonym** [comment]
its extremely easy just cure my depression lol

**Comment by abyssonym** [comment]
alright v2 is up, it should be fixed now
